controllers/etcdcluster_controller.go

```diff
@@ -84,29 +86,55 @@ func (r *EtcdClusterReconciler) updateStatus(
 				ID:   member.ID,
 			}
 		}
-
-		if err := r.Client.Status().Update(ctx, cluster); err != nil {
-			return err
+	} else {
+		if cluster.Status.Members == nil {
+			cluster.Status.Members = make([]etcdv1alpha1.EtcdMember, 0)
```

Comment:
Why do we do this? We can leave this as nil if it contains nothing?

Reply:
Seemed polite to add an empty list and not leave a `nil` dangling around. This also means that after the first reconcile this should never be `nil`.